plugins/plextor/qscan_plugin_plextor.cpp

```diff
@@ -0,0 +1,263 @@
+  dev = idev;
+  if (!dev->silent)
+    printf("scan_plextor()\n");
+  devlist = (drivedesc *)&drivelist;
+  test = 0;
+}
+
+scan_plextor::~scan_plextor() {
+  if (!dev->silent)
+    printf("~scan_plextor()\n");
+}
+
+int scan_plextor::probe_drive() {
+  if (isPlextor(dev))
+    plextor_px755_do_auth(dev);
+  if (dev->media.type & DISC_CD) {
+    if (cmd_cd_errc_init())
+      return DEV_FAIL;
+    if (cmd_scan_end())
+      return DEV_FAIL;
+  } else if (dev->media.type & DISC_DVD) {
+    if (cmd_dvd_errc_init())
+      return DEV_FAIL;
+    if (cmd_scan_end())
+      return DEV_FAIL;
+  } else {
+    return DEV_FAIL;
+  }
+  return DEV_PROBED;
+}
+
+int scan_plextor::errc_data() {
+  if (dev->media.type & ~DISC_DVDRAM)
+    return 0;
+
+  if (dev->media.type & DISC_CD) {
+    return (ERRC_DATA_BLER | ERRC_DATA_E11 | ERRC_DATA_E21 | ERRC_DATA_E31 |
+            ERRC_DATA_E12 | ERRC_DATA_E22 | ERRC_DATA_E32 | ERRC_DATA_UNCR);
+  } else if (dev->media.type & DISC_DVD) {
+    return (ERRC_DATA_PIE | ERRC_DATA_PIF | ERRC_DATA_POE | ERRC_DATA_POF |
+            ERRC_DATA_UNCR);
+  }
+  return 0;
+}
+
+int scan_plextor::check_test(unsigned int itest) {
+  switch (itest) {
+  case CHK_ERRC:
+  case CHK_JB:
+    if (dev->media.type & ~DISC_DVDRAM)
+      return 0;
+    break;
+  case CHK_FETE:
+    if (dev->media.type & (DISC_CD | DISC_DVD) & ~DISC_CDROM & ~DISC_DVDROM)
+      return 0;
+    break;
+  case CHK_TA:
+    if (dev->media.type & (DISC_DVD & ~DISC_DVDROM))
+      // if (dev->media.type & DISC_DVD)
+      return 0;
+    break;
+  default:
+    break;
+  }
+  return -1;
+}
+
+int *scan_plextor::get_test_speeds(unsigned int itest) {
+  switch (itest) {
+  case CHK_ERRC:
+    if (dev->media.type & DISC_CD)
+      return (int *)SPEEDS_ERRC_CD;
+    if (dev->media.type & DISC_DVD)
+      return (int *)SPEEDS_ERRC_DVD;
+    break;
+  case CHK_JB:
+    if (dev->media.type & DISC_CD)
+      return (int *)SPEEDS_JB_CD;
+    if (dev->media.type & DISC_DVD)
+      return (int *)SPEEDS_JB_DVD;
+    break;
+  default:
+    break;
+  }
+  return NULL;
+}
+
+int scan_plextor::start_test(unsigned int itest, long ilba, int &speed) {
+  int r = -1;
+  plextor_px755_do_auth(dev);
+  switch (itest) {
+  case CHK_ERRC_CD:
+    lba = ilba;
+    set_read_speed(speed);
+    r = cmd_cd_errc_init();
+    break;
+  case CHK_JB_CD:
+    lba = ilba;
+    set_read_speed(speed);
+    r = cmd_cd_jb_init();
+    break;
+  case CHK_ERRC_DVD:
+    lba = ilba;
+    set_read_speed(speed);
+    r = cmd_dvd_errc_init();
+    break;
+  case CHK_JB_DVD:
+    lba = ilba;
+    set_read_speed(speed);
+    r = cmd_dvd_jb_init();
+    break;
+#if 1
+  case CHK_FETE:
+#else
+  case CHK_FETE_CD:
+  case CHK_FETE_CDROM:
+  case CHK_FETE_DVD:
+  case CHK_FETE_DVDROM:
+#endif
+    lba = ilba;
+    set_write_speed(speed);
+    r = cmd_fete_init();
+    break;
+  case CHK_TA:
+    if (dev->media.type & (DISC_DVD & ~DISC_DVDROM)) {
+      r = 0;
```

Comment:
Codacy found an issue: [Variable 'r' is assigned a value that is never used.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6072101)